src/libs/relay/conduit_relay_io_blueprint.hpp

```diff
@@ -143,6 +143,61 @@ void CONDUIT_RELAY_API write_mesh(const conduit::Node &mesh,
                                   const std::string &protocol,
                                   const conduit::Node &opts);
 
+// -----------------------------------------------------------
+// get the number of local domains and the cycle info
+// -----------------------------------------------------------
+void CONDUIT_RELAY_API get_num_local_doms_and_cycle(int &local_num_domains,
+                                                    const Node &multi_dom,
+                                                    const bool &is_valid,
+                                                    std::string &opts_suffix,
+                                                    int &cycle,
+                                                    bool gen_name,
```

Comment:
We should break up this this function, it's doing more than its name suggests.
Also -- there are blueprint helpers to get the number of local domains -- can we use thoses?

In general, we want in params first and out params last in the list.